docs/operator-bundle/testing-and-releasing.md

```diff
@@ -0,0 +1,181 @@
+# Testing and releasing the operatorhub bundle for ASO
+
+This is a guide to producing and testing the operator bundle after an ASO release.
+
+## Test upgrading from the old bundle to the new one
+
+### Before you begin
+
+You'll need:
+
+* A previous operator hub bundle (for me this is `quay.io/operatorhubio/azure-service-operator:v0.37.0`)
+* A newly published operator image that needs wrapping in an operator bundle (eg. `mcr.microsoft.com/k8s/azureserviceoperator:1.0.22275`)
+* A cluster with OLM installed (but without ASO installed). Installing OLM:
+```sh
+curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.1/install.sh | bash -s v0.18.1
+```
+* An attached registry that you can push bundle and index images to (in my case this is `xtianregistry.azurecr.io`).
+  * If it's an ACR then log into it with:
+```sh
+az login
+az acr login --name <registry-name>
+```
+* The `opm` tool to build the index image:
+  * Follow [these instructions](https://docs.openshift.com/container-platform/4.5/operators/admin/olm-managing-custom-catalogs.html#olm-installing-opm_olm-managing-custom-catalogs) to get the `opm` binary.
+  * You can use docker rather than podman to log into the redhat registry, but you'll need to adjust the path to the auth credentials to match docker.
+  * The `oc image extract` command is convenient for getting the binary out of the right image layer. If you don't have/want oc installed you can find it manually using `docker image save` and `tar` to find and extract it.
+
+### Make the bundle image and index
+
```

Comment:
Suggest having an intro paragraph that describes what we want to achieve, and then launch into the numbered list detailing how.